library/data-model/src/data_storage/migrations/migrations.ts

```diff
@@ -154,14 +158,53 @@ export const invitesV1toV2Migration: MigrationFunc = doc => {
   return {action: 'update', updatedRecord: outputDoc};
 };
 
+export const invitesV2toV3Migration: MigrationFunc = doc => {
+  // Cast input document to V2 type
+  const inputDoc =
+    doc as unknown as PouchDB.Core.ExistingDocument<V2InviteDBFields>;
+
+  // Check for required fields
+  if (!inputDoc.projectId || !inputDoc.role) {
+    // If any required field is missing, abort and delete the document
+    return {action: 'delete'};
+  }
+
+  // Create the new V3 document structure
+  const outputDoc: PouchDB.Core.ExistingDocument<V3InviteDBFields> = {
+    // retain ID and rev
+    _id: inputDoc._id,
+    _rev: inputDoc._rev,
+    // Create a descriptive name
+    name: `${inputDoc.role} invite for ${inputDoc.projectId}`,
+    // Make them expire in one day from when this migration is applied
+    expiry: Date.now() + 24 * 60 * 60 * 1000,
+    // Project ID matches
+    resourceId: inputDoc.projectId,
+    // Invite for project
+    resourceType: Resource.PROJECT,
```

Comment:
If we happened to be migrating a V3 invite this would rewrite all invites as project invites, is it worth checking for inputDoc.resourceType to make this idempotent?

Reply:
defer this - version sniffer or other approach needed - consistent with other migrations